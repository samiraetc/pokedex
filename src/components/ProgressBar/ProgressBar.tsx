import React from "react";

export const COMPONENT_ID = 'pokedex-progress-bar'

type ProgressBarProps = {
  completed: number,
}

const ProgressBar = ({ completed }: ProgressBarProps) => {
  return (
    <div className="progress-bar" id={COMPONENT_ID}>
      <div className="progress-bar__container">
        <div className="progress-bar__filler" style={{ width: `${completed}%` }}>
        </div>
      </div>

      <span className="progress-bar__label">{completed}</span>
    </div>

  );
};

export default ProgressBar;