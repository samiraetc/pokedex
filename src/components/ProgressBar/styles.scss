.progress-bar {
    display: flex;
    align-items: center;
}

.progress-bar__container {
    height: $pokedex-spacing-20;
    width: 100%;
    background-color: #e0e0de;
    border-radius: 50px;
    display: flex;
    align-items: center;
    justify-content: space-between;
}

.progress-bar__filler {
    height: 100%;
    background-color: $pokedex-color-blue-strong;
    border-radius: inherit;
}

.progress-bar__label {
    color: black;
    font-weight: bold;
    font-size: $pokedex-font-size-14;
    margin: $pokedex-spacing-12;
}