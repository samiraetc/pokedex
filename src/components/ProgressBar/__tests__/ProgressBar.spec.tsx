import { render, configure, fireEvent } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom';
import ProgressBar, { COMPONENT_ID } from '../ProgressBar';

configure({
    testIdAttribute: 'id',
  });
  
describe('[PROGRESSBAR]', () => {
    test('should render ProgressBar', () => {
        const { asFragment, getByText, getByTestId } = render(<ProgressBar completed={80}/>);

        expect(getByText('80'));
        expect(getByTestId(COMPONENT_ID)).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

});