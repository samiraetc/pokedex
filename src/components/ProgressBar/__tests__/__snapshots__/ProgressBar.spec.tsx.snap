// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[PROGRESSBAR] should render ProgressBar 1`] = `
<DocumentFragment>
  <div
    class="progress-bar"
    id="pokedex-progress-bar"
  >
    <div
      class="progress-bar__container"
    >
      <div
        class="progress-bar__filler"
        style="width: 80%;"
      />
    </div>
    <span
      class="progress-bar__label"
    >
      80
    </span>
  </div>
</DocumentFragment>
`;
